### PR TITLE
feat: allow specifying query parameters

### PR DESCRIPTION
**Pull request by guoweis-outreach**
This allows specifying the query parameters and properly encode them.

This is for issue #625

### CODE REVIEW

**Review by yesnault — CHANGES_REQUESTED** [review]
Hi @guoweis-outreach , thank you for this PR.

Can you add documentation on http/README.md file, and  unit test and/or tests into tests/ directory about that please?

Thank you

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#1086.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by guoweis-outreach** [comment]
@yesnault i'm not sure how to run `Acceptance Tests` or where to find the log why `Acceptance Tests` failed. Any tips?

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#33.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#37.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

File: tests/http.yml [hunk 1]
```diff
@@ -91,3 +91,13 @@ testcases:
...
+
+- name: get http (with query parameters)
+  steps:
+  - type: http
+    method: GET
+    url: https://httpbin.org/get
+    query_parameters:
+      a: " b"
+    assertions:
+      - result.bodyjson.args.b ShouldEqual " b"
```
Comment (yesnault) [inline tests/http.yml:103]:
```suggestion
      - result.bodyjson.args.a ShouldEqual " b"
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#43.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

**Pull request merged**